app/lib/api_support/forbid_params.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module ApiSupport
+
+  class UnpermittedParameters < StandardError
+    def initialize(unpermitted_keys)
+      msg = "Unpermitted parameters: #{unpermitted_keys}"
+      super msg
+    end
+  end
+
+  module ForbidParams
+    extend ::ActiveSupport::Concern
+    include Params
+
+    included do
+      class_attribute :_forbid_params_action
+      class_attribute :_forbid_params_whitelist, default: []
+
+      rescue_from UnpermittedParameters do |error|
+        render_error error.message, status: :unprocessable_entity
+      end
+    end
+
+    class_methods do
+      def forbid_extra_params(action, options = {})
+        before_action :unpermitted_parameters_check
```

Comment:
>This is standard for ActiveSupport::Concern

What is standard?

Reply:
`class_methods` is an `ActiveSupport::Concern` thing. But now I realize you're talking about `before_action :unpermitted_parameters_check`. :no_mouth: 

And I understand your concern. Joan demonstrated though that it's not the case and added a test case. Also we may raise in case `self._forbid_params_action` was already set to catch occasions where we call this method twice.

I suggested the approach because it is mostly clean for controllers that don't use it and only things change for controllers that do.

For example in present version, one can include the module in the base controller and then only call `forbid_extra_params` wherever needed.

It is very similar to how it is done in gems like [audited](https://github.com/collectiveidea/audited/blob/900303f7a296884957c27794eb01b4dac43d50d6/lib/audited/auditor.rb#L85-L88). And to my opinion it is nice.